### ISSUE TITLE
jsx-a11y/no-noninteractive-element-interactions shouldn't include label

### ISSUE THREAD

**Issue by RyanPWalker** [jsx-eslint/eslint-plugin-jsx-a11y#672]
Labels by nature are clickable and change/focus input all the time.  Why is it flagged for being noninteractive?

**Comment by jessebeach** [comment]
@RyanPWalker this is a great question. When you associate a label with a control, the agent extends the activation to the label for you. The label is not an inherently interactive element; it becomes interactive when paired with an interactive control. So, you should locate your handlers on the control element, not the label element. The agent will take care of the rest.

**Comment by anselmbradford** [comment]
@jessebeach Might you clarify/revisit this please? MDN includes `<label>` in interactive content https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/Content_categories#interactive_content

Simplified, I have the following structure:

```jsx
<div>
  <input id="toggle" type="checkbox" onKeyDown={keyDownActionFunc}>
  <label for="toggle" onMouseDown={mouseDownActionFunc}>
    Checkbox Label
  </label>
</div>
```

`eslint-plugin-jsx-a11y` flags the `onMouseDown` on the `label` as ` Non-interactive elements should not be assigned mouse or keyboard event listeners  jsx-a11y/no-noninteractive-element-interactions`.

When you click the label, the target is the label, not the input. If the mousedown is moved to the input, then clicking the label does not trigger the mousedown.

**Comment by ljharb** [comment]
So why not listen to the change event on the input instead of any key or mouse events?

**Comment by anselmbradford** [comment]
> So why not listen to the change event on the input instead of any key or mouse events?

Good thinking, this is actually what I ended up doing. 

Still seems like a discrepancy that `label` is in interactive content https://www.w3.org/TR/2011/WD-html5-author-20110809/content-models.html#interactive-content

**Comment by jessebeach** [comment]
> Still seems like a discrepancy that label is in interactive content https://www.w3.org/TR/2011/WD-html5-author-20110809/content-models.html#interactive-content

0_o, I'd never noticed this before. `<label>` isn't focusable, but it is sort of interactive. That's a very annoying exception to the type :/
